lib/ex_doc/formatter/html.ex

```diff
@@ -82,19 +88,45 @@ defmodule ExDoc.Formatter.HTML do
     end
   end
 
-  defp generate_readme(output, module_nodes, config, modules, exceptions, protocols) do
+  defp generate_readme(output, module_nodes, config, modules, exceptions, protocols, logo_path) do
     readme_path = Path.expand(config.readme)
-    write_readme(output, File.read(readme_path), module_nodes, config, modules, exceptions, protocols)
+    write_readme(output, File.read(readme_path), module_nodes, config, modules, exceptions, protocols, logo_path)
   end
 
-  defp write_readme(output, {:ok, content}, module_nodes, config, modules, exceptions, protocols) do
+  defp process_logo_metadata(config) do
```

Comment:
Can't we just assume that they are giving us the logo with the proper extension? Call `Path.extname/1`, it should return either `.jpg` or `.png`, otherwise we fail. :)